### PR TITLE
fix: fix validation on lab

### PR DESCRIPTION
**Pull request by luthfiazhari**
### JIRA Link
[https://blocksphere2020.atlassian.net/browse/DBIO-1361](url)

### Changelog / Description
- fix validation and bugs 
- fix price cannot 0
- fix expected duration cannot 0
- fix expected duration input (working day, day, hours)

### CODE REVIEW

File: src/views/Dashboard/Lab/Account/Certification.vue [hunk 1]
```diff
@@ -128,15 +128,15 @@
                 ></v-textarea>
                 <v-file-input
                     dense
-                    label="Supporting Document"
+                    label="Supporting Documentsss"
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Account/Certification.vue:131]: do you forgot to change this?

File: src/views/Dashboard/Lab/Account/Certification.vue [hunk 2]
```diff
@@ -128,15 +128,15 @@
...
                     placeholder="Supporting Document"
                     prepend-icon="mdi-file-document"
                     outlined
                     @change="fileUploadEventListener"
                     :rules="supportingDocumentsRules"
                     show-size
                     v-model="files"
-                    accept="application/pdf, image/png, image/jpeg,"
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Account/Certification.vue:139]: it doesn't support png/jpeg?

**Review by hilyds — COMMENTED** [review]
this doesn't resolve the DBIO-1361, but only field validation. please check some comment above

File: src/views/Dashboard/Lab/Account/Certification.vue (hunk 1 above)
Comment (luthfiazhari) [reply src/views/Dashboard/Lab/Account/Certification.vue]: yes my bad, i already fix this

File: src/views/Dashboard/Customer/EMR/UploadEMR.vue [hunk 3]
```diff
@@ -326,14 +326,14 @@ export default {
     },
     titleRules() {
       return [
-        val => !!val || 'Title is Required',
-        val => (val && val.length <= 50) || 'Max 50 Character'
+        val => !!val || 'This field is required',
+        val => (val && val.length <= 50) || 'This field only allows 255 characters'
```
Comment (hilyds) [inline src/views/Dashboard/Customer/EMR/UploadEMR.vue:330]: The validation is <=50, but the text ist 255 characters. Please fix which one is true

File: src/views/Dashboard/Doctor/Account/Certification.vue [hunk 4]
```diff
@@ -169,34 +169,34 @@ export default {
...
         file => !file || file.type == 'application/pdf' || 'Document type should be application/pdf',
-        file => !file || file.size <= 2_000_000 || 'Document size should be less than 2 MB!',
+        file => !file || file.size <= 2_000_000 || 'The total file size uploaded exceeds the maximum file size allowed (2MB)',
       ]
     },
     titleRules() {
       return [
-        val => !!val || 'Title is required',
+        val => !!val || 'This field is required',
         val => (val && val.length <= 50) || 'Max 50 Character'
```
Comment (hilyds) [inline src/views/Dashboard/Doctor/Account/Certification.vue:178]: should the message validation be the same : 'This field only allows 50 characters' ?

File: src/views/Dashboard/Doctor/Account/Certification.vue [hunk 5]
```diff
@@ -169,34 +169,34 @@ export default {
...
       ]
     },
     issuerRules() {
       return [
-        val => !!val || 'Issuer is required',
+        val => !!val || 'This field is required',
         val => (val && val.length <= 100) || 'Max 100 Character']
```
Comment (hilyds) [inline src/views/Dashboard/Doctor/Account/Certification.vue:184]: should the message validation be the same : 'This field only allows 100 characters' ?

File: src/views/Dashboard/Doctor/Account/Verification.vue [hunk 6]
```diff
@@ -48,7 +48,7 @@ export default {
   computed: {
     fileInputRules() {
       return [
-        value => !value || value.size < 2000000 || 'Image size should be less than 2 MB!',
+        value => !value || value.size < 2000000 || 'The total file size uploaded exceeds the maximum file size allowed (2MB',
```
Comment (hilyds) [inline src/views/Dashboard/Doctor/Account/Verification.vue:51]: miss the ')' at the end of the sentence

File: src/views/Dashboard/Doctor/Registration/index.vue [hunk 7]
```diff
@@ -160,25 +160,25 @@ export default {
     },
     emailRules() {
       return [
-        val => !!val || 'E-mail is required',
+        val => !!val || 'This field is required',
         val => /.+@.+\..+/.test(val) || 'E-mail must be valid',
```
Comment (hilyds) [inline src/views/Dashboard/Doctor/Registration/index.vue:164]: Make the error message consistent with other email field validation :
Email is invalid. It should contain @ followed by a domain

File: src/views/Dashboard/Hospital/Account/Certification.vue [hunk 8]
```diff
@@ -169,34 +169,34 @@ export default {
...
         file => !file || file.type == 'application/pdf' || 'Document type should be application/pdf',
-        file => !file || file.size <= 2_000_000 || 'Document size should be less than 2 MB!',
+        file => !file || file.size <= 2_000_000 || 'The total file size uploaded exceeds the maximum file size allowed (2MB)',
       ]
     },
     titleRules() {
       return [
-        val => !!val || 'Title is required',
+        val => !!val || 'This field is required',
         val => (val && val.length <= 50) || 'Max 50 Character'
```
Comment (hilyds) [inline src/views/Dashboard/Hospital/Account/Certification.vue:178]: Make the error message consistent with other field validation :
This field only allows 50 characters'

File: src/views/Dashboard/Hospital/Account/Certification.vue [hunk 9]
```diff
@@ -169,34 +169,34 @@ export default {
...
       ]
     },
     issuerRules() {
       return [
-        val => !!val || 'Issuer is required',
+        val => !!val || 'This field is required',
         val => (val && val.length <= 100) || 'Max 100 Character']
```
Comment (hilyds) [inline src/views/Dashboard/Hospital/Account/Certification.vue:184]: Make the error message consistent with other field validation :
This field only allows 100 characters'

File: src/views/Dashboard/Hospital/Registration/index.vue [hunk 10]
```diff
@@ -160,25 +160,25 @@ export default {
     },
     emailRules() {
       return [
-        val => !!val || 'E-mail is required',
+        val => !!val || 'This field is required',
         val => /.+@.+\..+/.test(val) || 'E-mail must be valid',
```
Comment (hilyds) [inline src/views/Dashboard/Hospital/Registration/index.vue:164]: Make the error message consistent with other email field validation :
Email is invalid. It should contain @ followed by a domain

File: src/views/Dashboard/Lab/Account/index.vue [hunk 11]
```diff
@@ -200,24 +200,24 @@ export default {
     
     rules(){
       return [
-        file => !file || file.size <= 3_097_152 || 'Document size should be less than 3 MB!',
+        file => !file || file.size <= 3_097_152 || 'The total file size uploaded exceeds the maximum file size allowed (3MB)',
         file => !file || file.type == 'image/jpg' || file.type == 'image/jpeg' || 'Document type should be image/jpg',
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Account/index.vue:204]: should also mention 'jpeg' in the error message. also, png format should also supported

File: src/views/Dashboard/Lab/Registration/Services/index.vue [hunk 12]
```diff
@@ -263,7 +263,8 @@ export default {
       {text: 'Hours', value: 'Hours'},
       {text: 'Days', value: 'Days'}
     ],
-    selectExpectedDuration: {text: 'Working Days', value: 'WorkingDays'},
+    // selectExpectedDuration: {text: 'Working Days', value: 'WorkingDays'},
+    selectExpectedDuration: 'Working Day',
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Registration/Services/index.vue:267]: why change this line?

File: src/views/Dashboard/Lab/Registration/Services/index.vue [hunk 13]
```diff
@@ -467,6 +470,9 @@ export default {
     },
 
     async createService() {
+      console.log(this.expectedDuration, "expected duration")
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Registration/Services/index.vue:473]: need to remove this line and the line below

File: src/views/Dashboard/Lab/Registration/index.vue [hunk 14]
```diff
@@ -223,44 +223,45 @@ export default {
...
       return [
-        val => !!val || 'Address is Required',
-        val => (val && val.length <= 180) || 'Max 180 Characters'
+        val => !!val || 'This field is required',
+        val => (val && val.length <= 255) || 'This field only allows 255 characters'
       ]
     },
 
-    phoneNumberRules() {
+    phoneNumberRules() {// validasi di exel belom ada
```
Comment (hilyds) [inline src/views/Dashboard/Lab/Registration/index.vue:245]: the comment in here, use english next time. so it's still need to fix?

**Review by hilyds — CHANGES_REQUESTED** [review]
please do checkout some of my comments

**Pull request merged**